apps/api/src/app/events/usecases/trigger-event/trigger-event.usecase.ts

```diff
@@ -49,6 +52,15 @@ export class TriggerEvent {
       return this.logTemplateNotActive(command, template);
     }
 
+    const defaultPayload = this.verifyPayload.execute(
+      VerifyPayloadCommand.create({
+        payload: command.payload,
+        template,
+      })
+    );
+
+    command.payload = { ...command.payload, ...defaultPayload };
```

Comment:
In the case of nested object in one or the other, won't they cancel each other?

Reply:
True, nice finding! I thought the ES6 spread nowadays does more than just a shallow merge haha